futures-util/src/async_await/join.rs

```diff
@@ -1,5 +1,25 @@
-//! The `join` macro.
 
+/// The `join` macro.
+/// 
+/// This macro polls multiple futures simultaneously, returning a tuple
+/// of all results once complete. While `join!(a, b)` is similar to `(await!(a), await!(b))`,
+/// unlike the latter `join!` can make progress on multiple pending futures at once.
+/// 
+/// This macro is only usable inside of `async` functions, closures, and blocks.
```

Comment:
Highlight "async" only when you mean the keyword. (Same with `Future`, we usually write just "future"). I once submitted a PR to the async functions RFC that highlighted all the keywords. But it was rejected because it looked cluttered (and it did!) ^^'